### PR TITLE
Update broker deployments spec for better availability during upgrades

### PR DESCRIPTION
**Pull request by yolocs**
Fixes https://github.com/google/knative-gcp/issues/611

## Proposed Changes

- Add `MinReadySeconds=60s` for broker deployments
- Add a safer rolling update strategy
- Relax probe period

**Release Note**

```release-note
Update broker deployments spec for better availability during upgrades
- Add `MinReadySeconds=60s` for broker deployments
- Add a safer rolling update strategy
- Relax probe period
```

**Docs**

### CODE REVIEW

File: pkg/reconciler/brokercell/resources/deployments.go [hunk 1]
```diff
@@ -135,6 +135,13 @@ func deploymentTemplate(args Args, containers []corev1.Container) *appsv1.Deploy
 		},
 		Spec: appsv1.DeploymentSpec{
 			Selector: &metav1.LabelSelector{MatchLabels: Labels(args.BrokerCell.Name, args.ComponentName)},
+			Strategy: appsv1.DeploymentStrategy{
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxSurge:       &intstr.IntOrString{IntVal: 1},
+					MaxUnavailable: &intstr.IntOrString{IntVal: 0},
+				},
+			},
+			MinReadySeconds: 60,
```
Comment (Ectelion) [inline pkg/reconciler/brokercell/resources/deployments.go:144]: Interesting investigation on the task page!
Regarding MinReadySeconds, I wonder if this applies to upgrade only or does it also apply to new replicas added during autoscaling?

Comment (yolocs) [reply pkg/reconciler/brokercell/resources/deployments.go:144]: It should impact that too.

Comment (Ectelion) [reply pkg/reconciler/brokercell/resources/deployments.go:144]: I see. And there is no such option that is upgrade-specific? It's just in the context of our recent work on fanout where we rely increasingly more on autoscaling, I was worrying if an additional the scale out delay of 45 seconds may come at some cost.

Comment (yolocs) [reply pkg/reconciler/brokercell/resources/deployments.go:145]: I don't see any problem at the moment. This delay doesn't stop the pod from being ready and start pulling pubsub.

Comment (Ectelion) [reply pkg/reconciler/brokercell/resources/deployments.go:145]: Cool, I have misinterpreted available vs ready then.

**Pull request merged**